src/InjectHostDevBufferCopies.cpp

```diff
@@ -578,16 +597,7 @@ class InjectBufferCopies : public IRMutator {
             std::vector<Expr> create_buffer_args;
             internal_assert(buffer_init_let) << "Could not find definition of " << op->name << ".buffer\n";
 
-            const Call *possible_create_buffer = buffer_init_let->value.as<Call>();
-            if (possible_create_buffer != nullptr &&
-                possible_create_buffer->name == Call::buffer_init) {
-                // Use the same args, but with a zero host pointer.
-                create_buffer_args = possible_create_buffer->args;
-                create_buffer_args[1] = make_zero(Handle());
-            }
-
-            Expr buf = Call::make(type_of<struct buffer_t *>(), Call::buffer_init,
-                                  create_buffer_args, Call::Extern);
+            Expr buf = NullifyHostField(op->name).mutate(buffer_init_let->value);
```

Comment:
A comment indicating what kind of thing it was before nullifying would be good (a malloc, a pointer to another buffer that owns the allocation, ...). This is a bit suspicious as a possible source of memory leaks.